### PR TITLE
Filebeat Redis module

### PR DESCRIPTION
**Pull request by tsg**
Contains two filesets:

* One for collecting & parsing the logs written on disk by Redis
* One that uses the Redis prospector to interrogate the slow logs

TODOs:

* [x] docs & changelog
* [x] dashboards
* [x] test with redis authentication

This also moves `beat.read_timestamp` to `read_timestamp` to match the rest of the modules.

### CODE REVIEW

**Comment by tsg** [comment]
Sample dashboard:

2017-06-02-14-51-localhost-5601

File: filebeat/prospector/redis/harvester.go [hunk 1]
```diff
@@ -131,9 +131,7 @@ func (h *Harvester) Run() error {
 			"redis": common.MapStr{
 				"slowlog": subEvent,
 			},
-			"beat": common.MapStr{
-				"read_timestamp": common.Time(time.Now()),
-			},
+			"read_timestamp": common.Time(time.Now()),
```
Comment (ruflin) [inline filebeat/prospector/redis/harvester.go]: That means `read_timestamp` is now a "common" global field in filebeat?

**Comment by exekias** [comment]
LGTM, please address `check` error

File: filebeat/prospector/redis/harvester.go (hunk 1 above)
Comment (tsg) [inline filebeat/prospector/redis/harvester.go]: Yeah, that's how it is since we introduced the modules. We can still change it for 6.0..

Comment (ruflin) [inline filebeat/prospector/redis/harvester.go]: Yeah, now I remember we even had a quick discussion. +1 on changing it but not sure what to :-)

**Comment by tsg** [comment]
Remaining test failures don't look related.

**Pull request merged**